server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/utils/quotas/QuotaLoader.scala

```diff
@@ -0,0 +1,61 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.utils.quotas
+
+import eu.timepit.refined.api.Refined
+import javax.inject.Inject
+import org.apache.james.core.quota.{QuotaLimitValue, QuotaUsageValue}
+import org.apache.james.jmap.mail._
+import org.apache.james.jmap.model.UnsignedInt.UnsignedInt
+import org.apache.james.mailbox.exception.MailboxException
+import org.apache.james.mailbox.model.{MailboxPath, Quota => JavaQuota}
+import org.apache.james.mailbox.quota.QuotaManager
+
+class QuotaReader @Inject() (quotaManager: QuotaManager) {
+  @throws[MailboxException]
+  def retrieveQuotas(quotaRoot: QuotaRoot): Quotas = {
+    val quotaId: QuotaId = QuotaId.fromQuotaRoot(quotaRoot)
+
+    Quotas.from(
+      quotaId,
+      Quota.from(Map(
+        Quotas.Storage -> quotaToValue(quotaManager.getStorageQuota(quotaRoot.asJava)),
+        Quotas.Message -> quotaToValue(quotaManager.getMessageQuota(quotaRoot.asJava)))))
+  }
+
+  protected def quotaToValue[T <: QuotaLimitValue[T], U <: QuotaUsageValue[U, T]](quota: JavaQuota[T, U]): Value = {
+    Value(
+      Refined.unsafeApply(quota.getUsed.asLong),
+      quotaLimitValueToOptionUnsignedInt(quota.getLimit))
+  }
+
+  protected def quotaLimitValueToOptionUnsignedInt(value: QuotaLimitValue[_]): Option[UnsignedInt] = {
+    if (value.isUnlimited) {
+      return None
+    }
+    Some(Refined.unsafeApply(value.asLong))
```

Comment:
```
  /**
   * Wraps `t` in `[[Refined]][T, P]` ''without'' verifying that it satisfies
   * the predicate `P`. This method is for internal use only.
   */
```

You should not use this API but refineV.

The question is : what do you want to do if it fails?